ice/src/main/java/com/altinity/ice/cli/internal/cmd/InsertWatch.java

```
@@ -252,4 +340,25 @@ private static DeleteMessageBatchResponse deleteMessageBatch(
                     .toList())
             .build());
   }
+
+  private static SqsClient buildSqsClient(String sqsQueueURL) {
+    SqsClientBuilder builder = SqsClient.builder();
+
+    // Extract endpoint from queue URL for non-AWS endpoints (e.g., ElasticMQ)
+    // AWS SQS URLs look like: https://sqs.us-east-1.amazonaws.com/123456789012/queue-name
+    // ElasticMQ URLs look like: http://localhost:9324/000000000000/queue-name
+    try {
+      URI uri = URI.create(sqsQueueURL);
+      String host = uri.getHost();
+      if (host != null && !host.endsWith(".amazonaws.com")) {
+        URI endpoint = new URI(uri.getScheme(), null, host, uri.getPort(), null, null, null);
+        logger.info("Using custom SQS endpoint: {}", endpoint);
+        builder.endpointOverride(endpoint);
+      }
+    } catch (Exception e) {
+      logger.warn("Failed to parse SQS queue URL for endpoint extraction: {}", e.getMessage());
```

throw instead of ignore-logging, perhaps?